### DISCUSSION TITLE
Tying stub for Galaxy Integration API?

### DISCUSSION THREAD

**Discussion by ABaumher** [GOG-Nebula/.github#18] [Q&A]
I was running MyPy on the Steam Rework and I've run into the fact that the galaxy api package does not natively support linters (aka typing) as defined in PEP 561. I thought i saw a `.pyi` file for it somewhere but can't seem to find it anywhere. I'm in the process of making one, but don't want to do extra work if one is around somewhere. Has anyone here seen one?

@UncleGoogle @urwrstkn8mare @Mixaill

**Comment by urwrstkn8mare** [comment]
i'm not really that knowledgable here but doesn't this indicate that they use mypy as well? https://github.com/gogcom/galaxy-integrations-python-api/blob/6a1d2763c374bbcfea960310c7f81fbde131fbf0/mypy.ini

**Reply by ABaumher** [reply]
I think for it to be recognized when imported as a module in the plugin code, it needs a `py.typed` per PEP 561. So mypy in the plugin code complained it didn't have stubs for galaxy.

**Comment by UncleGoogle** [comment]
AFAIK there is no native stubs support for the API.

But in humble plugin I've used to run mypy in local venv with API pkg installed (as well on the CI) so that it can see all the code imported from there. The lib is small so I won't optimize it with if TYPE_CHECKING guard around import statement. Or maybe I misunderstand the usecase of pyi files?

**Reply by ABaumher** [reply]
My intention was just to make sure the code was without hidden issues when interacting with the api code. I can run mypy missing stubs for galaxy just fine but any calls to the galaxy api would silently pass without being checked. 

 Generating stubs for galaxy actually let me catch some bugs in the original code, it would set the error messasges to a dictionary when it meant to set the error data. Not a breaking bug since python doesn't enforce type checking but in theory could have caused issues
